ynput/ayon-maya
# Create ASS: Add more export options (AY-6100)

## Changelog Description

Expose more settings to the Create Ass export, like file type specific options for creating compressed `.ass.gz` exports.

## Additional info

New settings
```
ayon+settings://maya/create/CreateAss/maskImager    (default: false)
ayon+settings://maya/create/CreateAss/boundingBox    (default: true)
ayon+settings://maya/create/CreateAss/compressed    (default: false)
```
![image](https://github.com/user-attachments/assets/81bcc3a0-2026-473b-bb83-babb0b18295a)

Also on instance:
![image](https://github.com/user-attachments/assets/c61b9059-062f-42c4-9e5b-88a581f87ec9)

Using the settings as defaults.

### Question: Should I expose "Use Binary Encoding" ❓ 📔 

The Ass file type specific options also expose a "Use Binary Encoding" option:

![image](https://github.com/user-attachments/assets/a803480d-cee6-4a30-9891-5a0ed52dbed6)

I did not expose it on the Creator because it already seemed tied to a class variable on the Extractor [here](https://github.com/ynput/ayon-maya/blob/472458f8aea2ad15cb71b12c1e6bca08a13d08a2/client/ayon_maya/plugins/publish/extract_arnold_scene_source.py#L15). I wasn't entirely sure whether there was a reason for that - it did not seem to be exposed to settings. Looking at it again it seems that this is because [it's overridden in code for the proxy exports](https://github.com/ynput/ayon-maya/blob/5fd27f373d375ac6871eaec4c172a735c30ff0c5/client/ayon_maya/plugins/publish/extract_arnold_scene_source.py#L167)

@m-u-r-p-h-y Do we want to expose that too?
    - Also toggle-able on the instance? 
    - or should we keep the setting on the Extractor, also in settings hidden from the artists?

#### Improved validator

Also improved a validation report because I had no idea what it did when I hit the error - so I added some description to the report to be clearer for future me and other artists:

![image](https://github.com/user-attachments/assets/5b0db688-30c3-4ce4-99c0-a8412adac7fd)

## Testing notes:

1. Upload new maya addon package
3. Toggle settings should work
4. Export settings should work
5. Resulting publish should work and adhere to the instance settings
6. Loading the published instances should work as intended.

---

Clickup AY-6100
